### ISSUE TITLE
fn.max

### ISSUE THREAD

**Issue by gumowangfei** [dmlc/dgl#1331]
## 📚 Documentation
```python
import torch
import dgl
import dgl.function as fn

g = dgl.DGLGraph()
g.add_nodes(3)
g.add_edges([1, 2], [2, 2])

g.ndata['h'] = torch.tensor([[1.], [2.], [3.]])

def udf_u_e(edges):
    return {'m' : edges.src['h']}
def udf_max(nodes):
    print(nodes.mailbox['m'])
    return {'h_max' : torch.max(nodes.mailbox['m'], 1)[0]}

g.update_all(udf_u_e, udf_max) 
'''
g.ndata['h_max']
out:
tensor([[0.],
        [0.],
        [3.]])
'''
# g.update_all(udf_u_e, fn.max('m', 'h_max'))
g.update_all(fn.copy_u('h', 'm'), fn.max('m', 'h_max'))
'''
g.ndata['h_max']
out:
tensor([[-3.4028e+38],
        [-3.4028e+38],
        [ 3.0000e+00]])
'''
```
udf_max(...) and fn.max('m', 'h_max') output are inconsistent.

**Comment by jermainewang** [comment]
You are right. The default padding value for UDF is zero. To make them consistent, you need to manually set a feature initializer:
```python
import torch
import dgl
import dgl.function as fn

g = dgl.DGLGraph()
g.add_nodes(3)
g.add_edges([1, 2], [2, 2])

g.ndata['h'] = torch.tensor([[1.], [2.], [3.]])

def udf_u_e(edges):
    return {'m' : edges.src['h']}
def udf_max(nodes):
    print(nodes.mailbox['m'])
    return {'h_max' : torch.max(nodes.mailbox['m'], 1)[0]}

def initializer(shape, dtype, ctx, id_range):
    return torch.zeros(shape, dtype=dtype, device=ctx) + torch.finfo(dtype).min
g.set_n_initializer(initializer)
g.update_all(udf_u_e, udf_max)
print(g.ndata['h_max'])
# g.update_all(udf_u_e, fn.max('m', 'h_max'))
g.update_all(fn.copy_u('h', 'm'), fn.max('m', 'h_max'))
print(g.ndata['h_max'])
```
This gives:
```
tensor([[[2.],
         [3.]]])
tensor([[-3.4028e+38],
        [-3.4028e+38],
        [ 3.0000e+00]])
tensor([[-3.4028e+38],
        [-3.4028e+38],
        [ 3.0000e+00]])
```

**Comment by gumowangfei** [comment]
Thanks
